File: NetFramework/RulesManager/VB/ExcelLikeConditionalFormatting/Dialogs/BaseConditionalFormattingForm.vb
Imports System.ComponentModel

Public Class BaseConditionalFormattingForm
    <DefaultValue("Operation:")>
    Public Property ConditionalOperationText As String
        Get
            Return lblOperationText?.Text
        End Get
        Set(ByVal value As String)
            Dim operationText = lblOperationText

            If operationText IsNot Nothing Then
                operationText.Text = value
            End If
        End Set
    End Property

    <DefaultValue(DirectCast(Nothing, Object))>
    Public Property EditorRowControl As Control
        Get
            Return tlpMainLayout?.GetControlFromPosition(0, 1)
        End Get
        Set(ByVal value As Control)
            Dim mainLayout = tlpMainLayout

            If mainLayout Is Nothing Then
                Return
            End If

            Dim control = mainLayout.GetControlFromPosition(0, 1)

            If control IsNot Nothing Then

                If control IsNot value Then
                    mainLayout.Controls.Remove(control)
                Else
                    Return
                End If
            End If

            mainLayout.Controls.Add(value, 0, 1)
        End Set
    End Property
End Class

File: NetFramework/WinForms/CS/ControlExplorer/RulesManager/Dialogs/BaseConditionalFormattingForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Net.Http.Headers;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace ControlExplorer.RulesManager.Dialogs
{
    public partial class BaseConditionalFormattingForm : Form
    {
        public BaseConditionalFormattingForm()
        {
            InitializeComponent();
        }

        [DefaultValue("Operation:")]
        public string ConditionalOperationText
        {
            get
            {
                return lblOperationText?.Text;
            }
            set
            {
                var operationText = lblOperationText;
                if (operationText != null)
                {
                    operationText.Text = value;
                }
            }
        }

        [DefaultValue(null)]
        public Control EditorRowControl
        {
            get
            {
                return tlpMainLayout?.GetControlFromPosition(0, 1);
            }
            set
            {
                var mainLayout = tlpMainLayout;
                if (mainLayout == null)
                {
                    return;
                }

                var control = mainLayout.GetControlFromPosition(0, 1);
                if (control != null)
                {
                    if (control != value)
                    {
                        mainLayout.Controls.Remove(control);
                    }
                    else
                    {
                        return;
                    }
                }

                mainLayout.Controls.Add(value, 0, 1);
            }
        }

        private void BaseConditionalFormattingForm_Load(object sender, EventArgs e)
        {
            Explorer.ApplyTheme(this);
        }
    }
}
